src/psyclone/domain/lfric/kern_call_arg_list.py

```diff
@@ -126,10 +237,47 @@ def mesh_height(self, var_accesses=None):
         '''
         if self._kern.iterates_over not in ["cell_column", "domain"]:
             return
-        nlayers_name = self._symtab.find_or_create_tag("nlayers").name
-        self.append(nlayers_name, var_accesses)
+        nlayers_symbol = self.add_integer_reference("nlayers")
+        self.append(nlayers_symbol.name, var_accesses)
         self._nlayers_positions.append(self.num_args)
 
+    def scalar(self, scalar_arg, var_accesses=None):
+        '''
+        Add the necessary argument for a scalar quantity as well as an
+        appropriate Symbol to the SymbolTable.
+
+        :param scalar_arg: the scalar kernel argument.
+        :type scalar_arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        :param var_accesses: optional VariablesAccessInfo instance that \
+            stores information about variable accesses.
+        :type var_accesses: \
+            :py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        :raises NotImplementedError: if a scalar of type other than real \
+            or integer is found.
+
+        '''
+        super().scalar(scalar_arg, var_accesses)
+        if scalar_arg.is_literal:
+            literal = scalar_arg.name
+            if scalar_arg.intrinsic_type == "integer":
+                datatype = psyir.LfricIntegerScalarDataType()
+                # TODO #1919: there should be a better way to avoid
```

Comment:
I've used the SCALAR_PRECISION_MAP, but did not do any other test, since an error will be raised in the Literal anyway if there is a `_` in the real constant (after we have just removed all valid `_precision` sub-strings).